src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -403,8 +408,8 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
         disk.bursting_enabled = enable_bursting
     if edge_zone is not None:
         disk.extended_location = edge_zone
-    if security_type is not None:
-        disk.security_profile = {'securityType': security_type}
+    if security_type or secure_vm_disk_encryption_set:
+        disk.security_profile = {'securityType': security_type, 'secure_vm_disk_encryption_set_id': secure_vm_disk_encryption_set}
```

Comment:
The logic of `security_type` needs to maintain the original logic, and the `secure_vm_disk_encryption_set` parameter can only be passed in when `--security-type` is `ConfidentialVM_DiskEncryptedWithCustomerKey`